### PR TITLE
MGMT-7274: Add Last-Modified to file middleware

### PR DESCRIPTION
**Pull request by carbonin**
Allow callers to provide a Last-Modified time when creating a file
download middleware. This will allow specifying the infra-env
updated time for things like the discovery ignition.

This will allow the image service to use this same time for the image
Last-Modified time.

## List all the issues related to this PR

https://issues.redhat.com/browse/MGMT-7274

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

Deployed assisted service with the patch, created an infra-env, and downloaded the discovery ignition:

```
$ curl -vv "http://localhost:8090/api/assisted-install/v2/infra-envs/e01468c5-9563-462c-8e32-3c46840b3fad/downloads/files?file_name=discovery.ign"
*   Trying ::1:8090...
* Connected to localhost (::1) port 8090 (#0)
> GET /api/assisted-install/v2/infra-envs/e01468c5-9563-462c-8e32-3c46840b3fad/downloads/files?file_name=discovery.ign HTTP/1.1
> Host: localhost:8090
> User-Agent: curl/7.76.1
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Content-Disposition: attachment; filename="discovery.ign"
< Content-Length: 13329
< Content-Type: application/octet-stream
< Last-Modified: Fri, 22 Apr 2022 19:44:28 GMT
< Date: Fri, 22 Apr 2022 19:47:17 GMT
```

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @avishayt 
/cc @danielerez 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -3378,7 +3378,7 @@ func (b *bareMetalInventory) V2DownloadHostIgnition(ctx context.Context, params
 		return common.GenerateErrorResponder(err)
 	}
 
-	return filemiddleware.NewResponder(installer.NewV2DownloadHostIgnitionOK().WithPayload(respBody), fileName, contentLength)
+	return filemiddleware.NewResponder(installer.NewV2DownloadHostIgnitionOK().WithPayload(respBody), fileName, contentLength, nil)
```
Comment (danielerez) [inline internal/bminventory/inventory.go:3381]: Maybe worth using params.InfraEnvID to fetch the entity and add infraEnv.UpdatedAt also here? Or is it out of the scope for now?

Comment (carbonin) [reply internal/bminventory/inventory.go:3381]: I don't think the infraenv has anything to do with the content of the host ignition, right?
Maybe host updated_at would be useful, but also I don't think it matters.

Comment (danielerez) [reply internal/bminventory/inventory.go:3381]: Right, we use infraenv id just to fetch the host from db. But yeah, it's not that interesting.

**Pull request merged**